docs/modules/ROOT/pages/accounts.adoc

```diff
@@ -1,632 +1,146 @@
-As of Cairo v0.10.0, contract classes should be declared from an Account contract.
-
-2. <<validate,`\\__validate__`>> verifies the transaction signature before executing the transaction with `\\__execute__`.
-
-3. <<execute,`\\__execute__`>> acts as the state-changing entry point for all user interaction with any contract, including managing the account contract itself.
-That's why if you want to change the public key controlling the Account, you would send a transaction targeting the very Account contract:
-
-[,python]
-----
-await signer.send_transaction(
-    account,
-    account.contract_address,
-    'set_public_key',
-    [NEW_KEY]
-)
-----
-
-Or if you want to update the Account's L1 address on the `AccountRegistry` contract, you would
-
-[,python]
-----
-await signer.send_transaction(account, registry.contract_address, 'set_L1_address', [NEW_ADDRESS])
-----
-
-NOTE: You can read more about how messages are structured and hashed in the https://github.com/OpenZeppelin/cairo-contracts/discussions/24[Account message scheme  discussion].
-For more information on the design choices and implementation of multicall, you can read the https://github.com/OpenZeppelin/cairo-contracts/discussions/27[How should Account multicall work discussion].
-
-The `\\__validate__` and `\\__execute__` methods accept the same arguments; however, `\\__execute__` returns a transaction response:
-
-[,cairo]
-----
-func __validate__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*) {
-}
+TIP: For detailed information on the usage and implementation check the xref:/api/account.adoc[API Reference] section.
 
-func __execute__(
-    call_array_len: felt, call_array: AccountCallArray*, calldata_len: felt, calldata: felt*
-) -> (response_len: felt, response: felt*) {
-}
-----
+== Table of Contents
 
-Where:
+* <<basic_account,Basic Account>>
+** <<account_entrypoints,Account entrypoints>>
+** <<interface,Interface>>
+* <<standard_account_interface,Standard Account Interface>>
+** <<isrc6_interface,ISRC6 Interface>>
+** <<isrc5_interface,ISRC5 Interface>>
+* <<deploying_an_account,Deploying an Account>>
 
-* `call_array_len` is the number of calls.
-* `call_array` is an array representing each `Call`.
-* `calldata_len` is the number of calldata parameters.
-* `calldata` is an array representing the function parameters.
+== Basic Account
 
-NOTE: The scheme of building multicall transactions within the `\\__execute__` method will change once StarkNet allows for pointers in struct arrays.
-In which case, multiple transactions can be passed to (as opposed to built within) `\\__execute__`.
+As we mentioned, Accounts in Starknet are smart contracts, and so they can be deployed and interacted
+with like any other contract, and can be extended to implement any custom logic. However, the Account is a special
+type of contract that is used to validate and execute transactions, and there are a set of entrypoints
```

Comment:
Both are entrypoints. They work together, but they are used separately. validate can even write to storage by itself.

Reply:
they're not really used separately. users call `__execute__` and that triggers `__validate__` underneath, making `__execute__` the sole entrypoint. can `__validate__` be called independently? what is the purpose/advantage of calling it entrypoint instead of function?

in any case this is a very small detail, no need to spend too much time with it.

Reply:
i guess what i didn't see is the purpose/advantage of calling it entrypoint instead of function, hence my suggestion.

are they used separately though? i think users call `__execute__` which triggers `__validate__` underneath, making `__execute__` the sole entrypoint. or can `__validate__` be called independently and with a real purpose?

Reply:
Validate can be called independently, not sure what you mean by real purpose. If the issue is with the word entrypoint in this context, I guess we need to define then what are we calling entrypoint. I'm using the word basically because it is what I recall the Starknet doc was using, and to me, it made sense (they are entrypoints for the sequencer to use depending on the context).